pkg/controllers/cluster/actions/replace.go

```diff
@@ -174,25 +175,28 @@ func (a *RackReplaceNode) replaceNode(ctx context.Context, state *State, member
 	)
 
 	a.Logger.Debug(ctx, "Waiting for PVC deletion", "member", member.Name, "pvc", pvc.Name)
-	if err := wait.PollImmediate(time.Second, time.Minute, func() (done bool, err error) {
-		p := &corev1.PersistentVolumeClaim{}
-		if err := cc.Get(ctx, naming.NamespacedName(naming.PVCNameForPod(member.Name), member.Namespace), p); err != nil {
-			if apierrors.IsNotFound(err) {
-				a.Logger.Debug(ctx, "PVC deleted", "member", member.Name, "pvc", pvc.Name)
-				return true, nil
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		return wait.PollImmediate(time.Second, time.Minute, func() (done bool, err error) {
+			p := &corev1.PersistentVolumeClaim{}
+			if err := cc.Get(ctx, naming.NamespacedName(naming.PVCNameForPod(member.Name), member.Namespace), p); err != nil {
+				if apierrors.IsNotFound(err) {
+					a.Logger.Debug(ctx, "PVC deleted", "member", member.Name, "pvc", pvc.Name)
+					return true, nil
+				}
+				return false, errors.Wrap(err, "failed to get pvc")
 			}
-			return false, errors.Wrap(err, "failed to get pvc")
-		}
 
-		// Remove finalizer, which will wait until pod is deleted.
-		// We want to delete pod anyway, so it's better to delete pvc immediately.
-		pvcCopy := p.DeepCopy()
-		pvcCopy.SetFinalizers([]string{})
-		if err := cc.Update(ctx, pvcCopy); err != nil {
-			return false, errors.Wrap(err, "failed to update pvc")
-		}
+			// Remove finalizer, which will wait until pod is deleted.
+			// We want to delete pod anyway, so it's better to delete pvc immediately.
+			pvcCopy := p.DeepCopy()
+			pvcCopy.SetFinalizers([]string{})
```

Comment:
might worth to check is there are any finalizers, otherwise you may not need the api call